spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java

```diff
@@ -97,14 +99,19 @@ CompositeSpanHandler compositeSpanHandler(ObjectProvider<SpanExportingPredicate>
 
 	@Bean
 	@ConditionalOnMissingBean
-	public Tracing braveTracing(Environment environment, List<SpanHandler> spanHandlers,
+	public Tracing braveTracing(Environment environment, TracingProperties properties, List<SpanHandler> spanHandlers,
 			List<TracingCustomizer> tracingCustomizers, CurrentTraceContext currentTraceContext,
 			Factory propagationFactory, Sampler sampler) {
+		if (properties.getPropagation().getType() == PropagationType.W3C
+				&& properties.getBrave().isSpanJoiningSupported()) {
+			throw new IncompatibleConfigurationException("management.tracing.propagation.type=w3c",
```

Comment:
Without changing `IncompatibleConfigurationException`, I think we should just pass in the keys here. While passing in key-value pairs works fine now, the failure analyser may be changed in the future and the error message would no longer read correctly.